packages/wavs/src/subsystems/trigger/streams/evm_stream/client/connection.rs

```diff
@@ -41,11 +47,17 @@ use super::{
 // 6. Try C (fail) → sleep 1s
 // 7. Try A (success)
 // ... and so on
-#[allow(dead_code)]
 pub struct Connection {
     handles: Option<[tokio::task::JoinHandle<()>; 2]>,
-    current_endpoint: Arc<std::sync::RwLock<Option<String>>>,
+    current_endpoint: Arc<RwLock<Option<String>>>,
     shutdown_txs: Option<[oneshot::Sender<()>; 2]>,
+    health_check_handle: Option<tokio::task::JoinHandle<()>>,
+    health_shutdown_tx: Option<oneshot::Sender<()>>,
+}
+
+pub struct PriorityEndpoint {
+    pub index: usize,
+    pub chain_key: ChainKey,
```

Comment:
I don't think this belongs here. We should either:

1. Make `chain_key` a first-class citizen on this whole client and put it in all the tracing logs

2. Remove the `chain_key` completely and health errors will only display the endpoint

Since this client is specific to WAVS altogether, I'd opt for (1)... but it will make this PR fairly noisy, so if you agree with that path please do in a separate PR

Reply:
https://github.com/Lay3rLabs/WAVS/pull/1046 went with option 1 here